noir-projects/noir-contracts/contracts/auth_contract/src/test/main.nr

```diff
@@ -0,0 +1,119 @@
+use crate::test::utils;
+use crate::Auth;
+
+use dep::aztec::prelude::AztecAddress;
+
+#[test]
+unconstrained fn main() {
+    // Setup without account contracts. We are not using authwits here, so dummy accounts are enough
+    let (env, auth_contract_address, admin, to_authorize, other) = utils::setup();
+
+    let authorized_is_unset_initially = || {
+        env.impersonate(admin);
+        let authorized = env.call_public(Auth::at(auth_contract_address).get_authorized());
+        assert_eq(authorized, AztecAddress::from_field(0));
+    };
+    authorized_is_unset_initially();
+
+    let non_admin_cannot_set_unauthorized = || {
+        env.impersonate(other);
+        env.assert_public_call_fails(Auth::at(auth_contract_address).set_authorized(to_authorize));
+    };
+    non_admin_cannot_set_unauthorized();
+
+    let admin_sets_authorized = || {
+        env.impersonate(admin);
+        env.call_public(Auth::at(auth_contract_address).set_authorized(to_authorize));
+        env.advance_block_by(1);
+
+        let scheduled_authorized = env.call_public(Auth::at(auth_contract_address).get_scheduled_authorized());
+        assert_eq(scheduled_authorized, to_authorize);
+    };
+    admin_sets_authorized();
+
+    let authorized_is_not_yet_effective = || {
+        env.impersonate(to_authorize);
+        let authorized: AztecAddress = env.call_public(Auth::at(auth_contract_address).get_authorized());
+        assert_eq(authorized, AztecAddress::zero());
+
+        env.assert_private_call_fails(Auth::at(auth_contract_address).do_private_authorized_thing());
+    };
+    authorized_is_not_yet_effective();
+
+    let authorized_becomes_effective_after_delay = || {
+        env.impersonate(to_authorize);
+
+        // We advance block by 4, because we initially advanced block by one after setting the value. See below comment for explanation.
+        env.advance_block_by(4);
```

Comment:
Let's make the contract's delay be a constant in this file. This comment is quite cryptic if you are not also looking at the contract source at the same time.